pkg/apiserver/experiment/experiment.go

```diff
@@ -144,8 +135,64 @@ func (s *Service) listExperiments(c *gin.Context) {
 // TODO: need to be implemented
 func (s *Service) getExperimentDetail(c *gin.Context) {}
 
-// TODO: need to be implemented
-func (s *Service) deleteExperiment(c *gin.Context) {}
+// @Summary Delete the specified chaos experiment.
+// @Description Delete the specified chaos experiment.
+// @Tags experiments
+// @Produce json
+// @Param namespace path string true "namespace"
+// @Param name path string true "name"
+// @Param kind query string true "kind" Enums(PodChaos, IoChaos, NetworkChaos, TimeChaos, KernelChaos, StressChaos)
+// @Success 200 "delete ok"
+// @Router /api/experiments/{namespace}/{name} [delete]
+// @Failure 400 {object} utils.APIError
+// @Failure 500 {object} utils.APIError
+func (s *Service) deleteExperiment(c *gin.Context) {
+	kind := c.Query("kind")
+	if kind == "" {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInvalidRequest.New("kind must be set"))
+		return
+	}
+	ns, name := c.Param("namespace"), c.Param("name")
+
+	ctx := context.TODO()
+	chaosKey := types.NamespacedName{Namespace: ns, Name: name}
+
+	var (
+		chaosKind *v1alpha1.ChaosKind
+		ok        bool
+	)
+	if chaosKind, ok = v1alpha1.Kinds[kind]; !ok {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInvalidRequest.New(kind + " is not supported"))
+		return
+	}
+	if err := s.kubeCli.Get(ctx, chaosKey, chaosKind.Chaos); err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
```
c.Status(http.StatusNotFound)
_ = c.Error()
```

Reply:
Regarding this comment https://github.com/pingcap/chaos-mesh/pull/524#discussion_r428105456. WDYT? Do you want to treat not found case as an error?

Reply:
Yes, I think so.